storm-client/src/jvm/org/apache/storm/windowing/Window.java

```diff
@@ -47,9 +47,17 @@
     List<T> getExpired();
 
     /**
-     * If processing based on event time, returns the watermark time otherwise the current timestamp.
+     * If processing based on event time, returns the window end time based on watermark otherwise
+     * returns the current timestamp.
```

Comment:
Thanks for noticing. Its actually returning the timestamp at which window ends for both event and processing time. I am invoking `System.currentTimeMillis` in `getEndTimestamp` which is un-necessary. Will reword the doc and refactor a bit to make it clear.